lib/services/index.js

```diff
@@ -133,24 +133,33 @@ function registerService(serviceName, opts, init) {
 
     debug('[%s] register internal service with alias %s from source %s', serviceName, alias, file);
 
-    this.list[alias] = new (require('./' + file))(this.kuzzle, opts, services[serviceName]);
+    this.list[alias] =
+      new (require('./' + file))(this.kuzzle, opts, services[serviceName]);
 
-    if (!init) {
-      return Bluebird.resolve();
+    if (! init) {
+      return Promise.resolve();
     }
 
-    return this.list[alias].init()
-      .timeout(timeout, '___timedOut')
-      .catch(err => {
-        if (err.message === '___timedOut') {
-          throw new KuzzleInternalError(`[FATAL] Service "${alias}[${serviceName}]" failed to init within ${timeout}ms`);
-        }
-
-        throw err;
-      });
+    return new Promise((resolve, reject) => {
+      setTimeout(() => {
+        const error = errorsManager.getError(
+          'external',
+          'common',
+          'service_initialization_timeout',
+          alias,
+          serviceName,
+          timeout);
+
+        reject(error);
+      }, timeout);
+
+      this.list[alias].init()
```

Comment:
You should add a finally section clearing the setTimeout above. Otherwise this promise will always be rejected after the timeout, even if it has already been resolved or rejected.